### PR TITLE
Feature : Biometric KYC

### PR DESCRIPTION
**Pull request by jumaallan**
Story: N/A

## Summary

Added Biometric KYC on Flutter

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -87,26 +89,29 @@ class MainContent extends StatelessWidget {
...
+                userId: userId,
+              ))
+              .then(
+                  (authResponse) => {
+                        SmileID.api.doEnhancedKycAsync(FlutterEnhancedKycRequest(
+                            country: "GH",
+                            idType: "DRIVERS_LICENSE",
+                            idNumber: "B0000000",
+                            callbackUrl:
+                                "https://webhook.site/a3d19f24-769a-46f2-997c-d186c3ae70ea",
```
[the remark this answers is not in the document]
Comment (jumaallan) [reply example/lib/main.dart:104]: Let me remove it. Its a generated url I use to test callbacks

File: example/lib/main.dart [hunk 2]
```diff
@@ -76,7 +77,8 @@ class MainContent extends StatelessWidget {
         enhancedKycAsyncButton(),
         documentVerificationButton(context),
         smartSelfieEnrollmentButton(context),
-        smartSelfieAuthenticationButton(context)
+        smartSelfieAuthenticationButton(context),
+        biometricKYCButton(context)
```
Comment (vanshg) [inline example/lib/main.dart:81]: Let's match the style used above with `enhancedKyc`
```suggestion
        biometricKycButton(context)
```

File: ios/Classes/SmileIDBiometricKYC.swift [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        _view = UIView()
+        _channel = FlutterMethodChannel(
+            name: "\(SmileIDBiometricKYC.VIEW_TYPE_ID)_\(viewId)",
+            binaryMessenger: messenger
+        )
+        _childViewController = nil
+        super.init()
+        let screen = SmileID.biometricKycScreen(
+            idInfo: IdInfo(
+                country:  args["userId"] as? String ?? "",
```
Comment (vanshg) [inline ios/Classes/SmileIDBiometricKYC.swift:28]: This should be `args["country"]`

File: ios/Classes/SmileIDBiometricKYC.swift [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                idType:  args["idType"] as? String,
+                idNumber:  args["idNumber"] as? String,
+                firstName:  args["firstName"] as? String,
+                middleName:  args["middleName"] as? String,
+                lastName:  args["lastName"] as? String,
+                dob:  args["dob"] as? String,
+                bankCode:  args["bankCode"] as? String,
+                entered:  args["entered"] as? Bool
+            ),
+            userId: args["userId"] as? String ?? "user-\(UUID().uuidString)",
```
Comment (vanshg) [inline ios/Classes/SmileIDBiometricKYC.swift:38]: Let's use the helper methods that are available in the iOS SDK
```suggestion
            userId: args["userId"] as? String ?? generateUserId(),
```

File: ios/Classes/SmileIDBiometricKYC.swift [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            jobId: args["jobId"] as? String ?? "job-\(UUID().uuidString)",
```
Comment (vanshg) [inline ios/Classes/SmileIDBiometricKYC.swift:39]: same here, `generateJobId()`

File: lib/smile_id_biometric_kyc.dart [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    bool showInstructions = true,
+    Map<String, String>? extraPartnerParams,
+    required Function(String resultJson) onSuccess,
+    required Function(String errorMessage) onError,
+  }) {
+    return SmileIDBiometricKYC._(
+      onSuccess: onSuccess,
+      onError: onError,
+      creationParams: {
+        "idType": idType,
```
Comment (vanshg) [inline lib/smile_id_biometric_kyc.dart:49]: We're missing `country` here

**Review by vanshg — CHANGES_REQUESTED** [review]

File: example/lib/main.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (vanshg) [reply example/lib/main.dart]: It was left in intentionally in the sample app as a way to verify that the callback is delivered

File: example/lib/main.dart (hunk 2 above)
Comment (jumaallan) [reply example/lib/main.dart:81]: Done

File: ios/Classes/SmileIDBiometricKYC.swift (hunk 4 above)
Comment (jumaallan) [reply ios/Classes/SmileIDBiometricKYC.swift:38]: Done

File: ios/Classes/SmileIDBiometricKYC.swift (hunk 5 above)
Comment (jumaallan) [reply ios/Classes/SmileIDBiometricKYC.swift:39]: Done

File: lib/smile_id_biometric_kyc.dart (hunk 6 above)
Comment (jumaallan) [reply lib/smile_id_biometric_kyc.dart:49]: Good catch - fixed

File: example/lib/main.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (jumaallan) [reply example/lib/main.dart]: Someone can see data posted on it - so we can remove it from the code and add it when testing only
